iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/string.inl

```diff
@@ -131,14 +132,17 @@ inline string<Capacity>::string(TruncateToCapacity_t, const char* const other, c
     }
     else if (Capacity < count)
     {
-#if defined(__GNUC__) && __GNUC__ == 8 && __GNUC_MINOR__ == 3
+// AXIVION DISABLE STYLE AutosarC++19_03-A16.0.1: conditional compilation is required for setting gcc diagnostics
```

Comment:
I think we should state why gcc 8 incorrectly warns here and therefore we chose the conditional compilation. On the other hand I'm not sure if maybe remove this constructor soon, see #1767 ...

Reply:
@FerdinandSpitzschnueffler should I keep this suppression for now? once the c'tor removed this may revisit, based on this #1767

Reply:
Since the discussion is still ongoing and I'm quite sure that we will forget to revisit if it will be decided to keep the constructor, I would propose to extend the justification to
``` conditional compilation is required for setting gcc diagnostics since ggc 8 incorrectly warns here about out of bounds array access```